File: fru1tworld/leetcode/~2000/leet1732/leet1732.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        vector<int> v(gain.size()+1, 0);

        for(int i=1; i<=gain.size(); i++){
            v[i] = v[i-1] + gain[i-1];
        }
        int ans = *max_element(v.begin(), v.end());
        return ans;
    }
};

File: iam454/week24/leet1732.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function (gain) {
  let point = 0;
  let ans = 0;

  for (let p of gain) {
    point += p;
    ans = Math.max(point, ans);
  }

  return ans;
};
